datadog_lambda/trigger.py

```diff
@@ -0,0 +1,182 @@
+    if source == "api-gateway":
+        request_context = event.get("requestContext")
+        return "arn:aws:apigateway:{}::/restapis/{}/stages/{}".format(
+            region, request_context["apiId"], request_context["stage"]
+        )
+
+    # e.g. arn:aws:logs:us-west-1:123456789012:log-group:/my-log-group-xyz
+    if source == "cloudwatch-logs":
+        with gzip.GzipFile(
+            fileobj=BytesIO(base64.b64decode(event["awslogs"]["data"]))
+        ) as decompress_stream:
+            data = b"".join(BufferedReader(decompress_stream))
+        logs = json.loads(data)
+        log_group = logs.get("logGroup", "cloudwatch")
+        return "arn:aws:logs:{}:{}:log-group:{}".format(region, account_id, log_group)
+
+    # e.g. arn:aws:events:us-east-1:123456789012:rule/my-schedule
+    if source == "cloudwatch-events" and event.get("resources"):
+        return event.get("resources")[0]
+
+
+def get_event_source_arn(source, event, context):
+    event_source_arn = read_event_source_arn(event)
+
+    event_record = get_first_record(event)
+    if event_record:
+        event_source_arn = read_event_source_arn(event_record)
+
+    if event_source_arn is None:
+        event_source_arn = parse_event_source_arn(source, event, context)
+
+    return event_source_arn
+
+
+def get_apigateway_http_tags(event):
+    """
+    Extracts HTTP tags from the triggering API Gateway event into HTTP facet tags
+    """
+    http_tags = {}
+    request_context = event.get("requestContext")
+    if request_context:
+        if request_context.get("domainName"):
+            http_tags["http.url"] = request_context["domainName"]
+
+        path = request_context.get("path")
+        method = request_context.get("httpMethod")
+        # Version 2.0 HTTP API Gateway
+        apigateway_v2_http = request_context.get("http")
+        if event.get("version") == "2.0" and apigateway_v2_http:
+            path = apigateway_v2_http.get("path")
+            method = apigateway_v2_http.get("method")
+
+        if path:
+            http_tags["http.url_details.path"] = path
+        if method:
+            http_tags["http.method"] = method
+
+    headers = event.get("headers")
+    if headers and headers.get("Referer"):
+        http_tags["http.referer"] = headers["Referer"]
+
+    return http_tags
+
+
+def get_trigger_tags(event, context):
+    """
+    Parses the trigger event object to get tags to be added to the span metadata
+    """
+    trigger_tags = {}
+    event_source = get_event_source(event)
+    if event_source:
+        trigger_tags["trigger.event_source"] = event_source
+
+        event_source_arn = get_event_source_arn(event_source, event, context)
+        if event_source_arn:
+            trigger_tags["trigger.event_source_arn"] = event_source_arn
+
+    if event_source == "api-gateway":
+        trigger_tags.update(get_apigateway_http_tags(event))
+
+    return trigger_tags
+
+
+def set_apigateway_status_code_tag(span, response):
```

Comment:
I think the status code might work for ALB as well, either way, I don't think it's necessary to call out apigateway, it might be better to be named as `set_http_status_code_tag`.

Reply:
BTW, did you look into ALB triggered Lambda? https://docs.aws.amazon.com/lambda/latest/dg/services-alb.html

Reply:
Ah, really good point. ALB/ELB totally skipped my mind. It's not in our P1 list but it'll be good to have anyways.